src/pe.rs

```diff
@@ -237,6 +237,20 @@ pub struct ImageVxdHeader {
     pub e32_ddkver: U16<LE>,
 }
 
+/// A PE rich header entry
+///
+/// Rich headers have no official documentation, but have been heavily reversed-engineered and documented in the wild, e.g.:
+/// * `http://www.ntcore.com/files/richsign.htm`
+/// * `https://www.researchgate.net/figure/Structure-of-the-Rich-Header_fig1_318145388`
+///
+/// These data are "masked", i.e. XORed with a rich header mask, that is file-specific.
+#[derive(Debug, Clone, Copy)]
+#[repr(C)]
+pub struct MaskedRichHeaderEntry {
+    pub masked_comp_id: U32<LE>,
+    pub masked_count: U32<LE>,
```

Comment:
Uh, I'm not sure what you mean here.

Do you want to ensure that "Rich" and "DanS" are aligned in the file?
Or do you want every `masked_comp_id` and `masked_count` to be aligned in memory (and/or in the file)? (and thus checking the "DanS" alignement is only a way to guarantee this) 

Besides, what's the difference between `U32` and `U32Bytes` regarding alignement? (if I understand correctly, one is just (?) an alias of the other `pub type U32<E> = U32Bytes<E>;`)

Reply:
`U32` is only an alias of `U32Bytes` when the `unaligned` feature is enabled.

When the `unaligned` feature is disabled, `data.read_slice_at::<pe::MaskedRichHeaderEntry>` will fail if the memory is not aligned to 4 bytes. Normally the memory is from `mmap`, so this is equivalent to requiring it to be aligned in the file.

Currently, `read_bytes_until_sequence` allows any alignment, not just a 4 byte alignment. Other rich header parsers that I have seen restrict this to a 4 byte alignment.

> Do you want to ensure that "Rich" and "DanS" are aligned in the file?
Or do you want every masked_comp_id and masked_count to be aligned in memory (and/or in the file)?

Aren't they equivalent requirements?

Reply:
OK, thanks for the tip, because the doc on docs.rs is generated as if the `unaligned` feature is set, and does not tell anything about how `U32` is implemented on the general case. I was quite confused but now I'm getting better.

I think it's OK to require 4 bytes alignment for these values, I'll push something that checks for unalignment earlier. However, that does not change the result, since the current implementation already properly returns an `Err` on unaligned values